sw/device/silicon_creator/manuf/base/perso_tlv_data.c

```diff
@@ -105,7 +108,10 @@ rom_error_t perso_tlv_cert_obj_build(const char *name, bool needs_endorsement,
   if (needs_endorsement) {
     PERSO_TLV_SET_FIELD(Objh, Type, obj_header, kPersoObjectTypeX509Tbs);
   } else {
-    PERSO_TLV_SET_FIELD(Objh, Type, obj_header, kPersoObjectTypeX509Cert);
+    // TODO(lowRISC/opentitan:#24281): should decide the obj_type by other
+    // factor
+    // PERSO_TLV_SET_FIELD(Objh, Type, obj_header, kPersoObjectTypeX509Cert);
+    PERSO_TLV_SET_FIELD(Objh, Type, obj_header, kPersoObjectTypeCwtCert);
```

Comment:
This will break call to this function in the ROM_EXT and perso fw where we actually want to generated CDI_0/1 X.509 certs that do not need endorsement: https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/silicon_creator/rom_ext/rom_ext.c;l=449?q=perso_tlv_cert_obj_build&ss=opentitan%2Fopentitan

Why don't we just pass in another parameter to `perso_tlv_cert_obj_build()` that is the cert type you want to build?

Reply:
Thanks, LGTM.